tools/fuzzer.cpp

```diff
@@ -0,0 +1,211 @@
+/// Copyright 2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+/// @author Copyright 2022, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include <iostream>
+#include <string>
+
+#include "velocypack/vpack.h"
+#include "velocypack/velocypack-exception-macros.h"
+#include <random>
+#include <thread>
+
+using namespace arangodb::velocypack;
+
+enum class Format {
+  VPACK, JSON
+};
+
+enum RandomBuilderAdditions {ADD_ARRAY=0, ADD_OBJECT, ADD_BOOLEAN, ADD_STRING, ADD_NULL, ADD_UINT64, ADD_INT64, ADD_DOUBLE};
+
+struct RandomGenerator {
+  RandomGenerator() : mt64{rd()} {}
+
+  std::random_device rd;
+  std::mt19937_64 mt64;
+};
+
+static void usage(char* argv[]) {
+  std::cout << "Usage: " << argv[0] << " [OPTIONS] [ITERATIONS]"
+            << std::endl;
+  std::cout << "This program creates <iterations> random VPack or JSON structures and validates them."
+            << std::endl;
+  std::cout << "The paralelization is supplied by <threads>."
+            << std::endl;
+  std::cout << "Available options are:" << std::endl;
+  std::cout << " --vpack       create VPack."
+            << std::endl;
+  std::cout << " --json        create JSON."
+            << std::endl;
+  std::cout << " <iterations>  number of iterations. Default: 1"
+            << std::endl;
+  std::cout << " <threads>  number of threads. Default: 1"
+            << std::endl;
+}
+
+static inline bool isOption(char const* arg, char const* expected) {
+  return (strcmp(arg, expected) == 0);
+}
+
+static void addString(Builder& builder, RandomGenerator& randomGenerator) {
+  static auto &availableChars = "0123456789"
+                                "abcdefghijklmnopqrstuvwxyz"
+                                "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  size_t length = randomGenerator.mt64() % 1000;
+  std::string s;
+  s.reserve(length);
+  while (length--) {
+    s += availableChars[randomGenerator.mt64() % (sizeof(availableChars) - 1)];
+  }
+  builder.add(Value(s));
+}
+
+static void generateVelocypack(Builder& builder, size_t depth, RandomGenerator& randomGenerator) {
+  RandomBuilderAdditions randomBuilderAdds;
+  while (true) {
+    randomBuilderAdds = static_cast<RandomBuilderAdditions>(randomGenerator.mt64() % 8);
+    if (depth > 10 && randomBuilderAdds < 2) {
```

Comment:
Now with the enum, all the magic numbers can be removed from the code.
```suggestion
    if (depth > 10 && randomBuilderAdds <= ADD_OBJECT) {
```